### PR TITLE
Amp live list pagination

### PR DESCRIPTION
**Pull request by erwinmombay**

### CODE REVIEW

File: build-system/server.js [hunk 1]
```diff
@@ -310,31 +338,59 @@ app.use(['/examples/*', '/extensions/*'], function (req, res, next) {
...
-    if (mode) {
-      file = file.replace(/(https:\/\/cdn.ampproject.org\/.+?).js/g, '$1.max.js');
-      file = file.replace('https://cdn.ampproject.org/v0.max.js', '/dist/amp.js');
-      file = file.replace(/https:\/\/cdn.ampproject.org\/v0\//g, '/dist/v0/');
-    }
-    if (mode == 'min') {
-      file = file.replace(/\.max\.js/g, '.js');
-      file = file.replace('/dist/amp.js', '/dist/v0.js');
-    }
+    file = replaceUrls(mode, file);
```
Comment (erwinmombay) [inline build-system/server.js]: @lannka can you review the changes to server.js? thanks

**Comment by erwinmombay** [comment]
@dvoytenko PTAL at amp-live-list pagination changes

File: build-system/server.js [hunk 2]
```diff
@@ -149,14 +149,28 @@ var liveListCtr = 0;
 var itemCtr = 2;
 var liveListDoc = null;
 var doctype = '<!doctype html>\n';
-app.use(liveListUpdateFile, function(req, res) {
+// Only handle min/max
+app.use('/examples/live-list-update.amp.(min.|max.)html', function(req, res) {
```
Comment (lannka) [inline build-system/server.js]: nit: 
`app.get`
`amp.(min|max).html`

File: build-system/server.js [hunk 3]
```diff
@@ -149,14 +149,28 @@ var liveListCtr = 0;
...
+  var filePath = req.baseUrl;
+  var mode = getPathMode(filePath);
+  // When we already have state in memory and user refreshes page, we flush
+  // the dom we maintain on the server.
+  if (!('amp_latest_update_time' in req.query) && liveListDoc) {
+    var outerHTML = liveListDoc.documentElement./*OK*/outerHTML;
+    outerHTML = replaceUrls(mode, outerHTML);
+    res.end(`${doctype}${outerHTML}`);
```
Comment (lannka) [inline build-system/server.js]: `res.send()` is better than `end()`.
it sets `content-type` header in response.

File: build-system/server.js [hunk 4]
```diff
@@ -149,14 +149,28 @@ var liveListCtr = 0;
...
     liveListDoc = jsdom.jsdom(liveListFile);
   }
   var action = Math.floor(Math.random() * 3);
   var liveList = liveListDoc.querySelector('#my-live-list');
+  var perPage = Number(liveList.getAttribute('data-max-items-per-page'));
+  var items = liveList.querySelector('[items]');
+  var pagination = liveListDoc.querySelector('#my-live-list [pagination]');
   var item1 = liveList.querySelector('#list-item-1');
   res.setHeader('Content-Type', 'text/html');
   res.statusCode = 200;
```
Comment (lannka) [inline build-system/server.js]: `setHeader` and `statusCode` not needed if you finally use `res.send(...)`

File: build-system/server.js (hunk 2 above)
Comment (erwinmombay) [inline build-system/server.js]: done.

File: build-system/server.js (hunk 3 above)
Comment (erwinmombay) [inline build-system/server.js]: done.

File: build-system/server.js (hunk 4 above)
Comment (erwinmombay) [inline build-system/server.js]: done. removed

**Comment by lannka** [comment]
LGTM

`server.js` is getting crowded. at some point, we might introduce a simple template engine in Express, instead of manipulate those HTMLs manually.

defer to @dvoytenko for the approval.

**Comment by erwinmombay** [comment]
@lannka yeah that would be good at some point

**Comment by dvoytenko** [comment]
LGTM

**Pull request merged**